core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -72,30 +77,37 @@ class ControllerChannelManager (private val controllerContext: ControllerContext
 
   def removeBroker(brokerId: Int) {
     brokerLock synchronized {
-      removeExistingBroker(brokerId)
+      removeExistingBroker(brokerStateInfo(brokerId))
     }
   }
 
   private def addNewBroker(broker: Broker) {
     val messageQueue = new LinkedBlockingQueue[(RequestOrResponse, (RequestOrResponse) => Unit)]()
-    debug("Controller %d trying to connect to broker %d".format(config.brokerId,broker.id))
-    val brokerEndPoint = broker.getBrokerEndPoint(config.interBrokerSecurityProtocol)
-    val channel = new BlockingChannel(brokerEndPoint.host, brokerEndPoint.port,
-      BlockingChannel.UseDefaultBufferSize,
-      BlockingChannel.UseDefaultBufferSize,
-      config.controllerSocketTimeoutMs)
-    val requestThread = new RequestSendThread(config.brokerId, controllerContext, broker, messageQueue, channel)
+    debug("Controller %d trying to connect to broker %d".format(config.brokerId, broker.id))
+    val selector = new Selector(
+      config.socketRequestMaxBytes,
+      config.connectionsMaxIdleMs,
+      metrics,
+      time,
+      //FIXME Replace random suffix by something better
+      s"controller-${config.brokerId}-broker-${broker.id}-${scala.util.Random.nextInt(1000)}",
```

Comment:
Yes, I understand that one selector per controller would be better. I was trying to avoid a complete redesign of `ControllerChannelManager` though. It currently uses one `RequestSendThread` per broker with a `BlockingQueue` and `BlockingChannel`. `ControllerChannelManager` has a public `sendRequest` method that adds to the relevant queue. `RequestSendThread` takes from this queue. My understanding is that a given Selector should be used from just one thread and hence the resulting code. Please let me know if this is not correct.

In an ideal world, we would redesign things to use one Selector per Controller, but it seemed riskier and I thought it would be better to do that in a separate change after the next release (since our main goal at this point is just to use `Selector` for the TLS/SSL support). What are your thoughts?